webpack/ForemanTasks/Components/TaskDetails/TaskDetailsReducer.js

```diff
@@ -1,32 +1,37 @@
 import Immutable from 'seamless-immutable';
+import { STATUS } from 'foremanReact/constants';
 import {
   FOREMAN_TASK_DETAILS_FETCH_TASK_REQUEST,
   FOREMAN_TASK_DETAILS_FETCH_TASK_SUCCESS,
   FOREMAN_TASK_DETAILS_STOP_POLLING,
   FOREMAN_TASK_DETAILS_START_POLLING,
+  FOREMAN_TASK_DETAILS_FETCH_TASK_FAILURE,
 } from './TaskDetailsConstants';
 
-const initialState = Immutable({});
+const initialState = Immutable({ isData: false });
 
 export default (state = initialState, action) => {
   const { type, payload } = action;
   let { taskReload, timeoutId } = state;
 
   switch (type) {
     case FOREMAN_TASK_DETAILS_FETCH_TASK_REQUEST:
-      return state.set('loading', true);
+      return state.set('status', STATUS.PENDING);
     case FOREMAN_TASK_DETAILS_FETCH_TASK_SUCCESS:
       if (payload.state === 'stopped') {
         clearTimeout(state.timeoutId);
         taskReload = false;
         timeoutId = null;
       }
       return state.merge({
-        loading: false,
+        status: STATUS.RESOLVED,
+        isData: true,
         ...payload,
         taskReload,
         timeoutId,
       });
+    case FOREMAN_TASK_DETAILS_FETCH_TASK_FAILURE:
+      return state.merge({ status: STATUS.ERROR, error: payload });
```

Comment:
you can use the API middleware where all of this logic already made for you,
and then just select the status and error from the state.